src/main/java/org/openjdk/jmc/ui/UIPlugin.java

```diff
@@ -0,0 +1,18 @@
+package org.openjdk.jmc.ui;
+
+import java.util.logging.Logger;
+
+/**
+ * TODO move this stub into -core This is only used for stubbing out a Logger instance used by JMC's
+ * JFCXMLValidator !!! This is a major classloading hackaround !!!
+ */
```

Comment:
@jiekang here's a good RFE for JMC - the JFCXMLValidator currently imports this UIPlugin class from another artifact, which is a subclass of an Eclipse AbstractUIPlugin class. So those UIPlugin classes are actually somehow related to the rendered application components and widgets, but clearly the XML Validator and the rest of the XML processing stuff I'm using here are not. And yet, the JMC code pathway for processing XML calls into UI components simply for the purpose of reusing their static logger instance...

A fix for this is probably going to take a while to make it into a snapshot release version so I'll have to rely on this hack for a while, but if any portion of this XML stuff can be cleaned up and moved into jmc-core, then it would be really nice and useful for us to use here in container-jfr(-core).

Reply:
Uh wow okay, that's quite a curiosity. This sounds like something warranting an issue opened in JMC Jira. If you agree I can write one up.

Reply:
Yea, I would agree with that. The XML stuff does currently live within a `ui` maven module, although within a `model` package, so that part makes sense. But IMO it's definitely wrong for a model to be using a view or controller's logger.

Reply:
And it's even worse than that really, because it isn't even the specific parent controller instance's logger or something - it's just a general static one on a class named UIPlugin...

Reply:
I think I can work around this in a slightly less ugly way by using reflection to make some constructors visible and then not use the JFCXMLValidator at all. This would let us use JMC's XML parsing, which is nice, but JFC validation would be great to retain - especially when I get to #143 and users can upload their own .jfc files for defining new templates.

Maybe I can do those reflection tricks plus implement a JFC validator based on the JMC one that doesn't call out to a UIPlugin logger and pass that in rather than allowing the JMC classes to default to their own validator.

Reply:
Nevermind - the `IXMLValidator` interface I'd need to implement is package-private, so making my own implementation doesn't really work. And the `JFCXMLValidator` class implementation is `final`. Well then.